### PR TITLE
Improve Jenkins library and merge package tests

### PR DESCRIPTION
**Pull request by CyanDevs**
# Changed Linux Tests
* Packaging `Ubuntu 20.04 SGX1FLC Package Debug LVI` -> Azure-Linux `Package ACC2004 Debug ControlFlow`. This is a new test, and will run only in the full test suite.
* Packaging `Ubuntu 20.04 SGX1FLC Package RelWithDebInfo LVI` -> Azure-Linux `Package ACC2004 RelWithDebInfo ControlFlow`. This is a pre-existing test that has been renamed from `Package ACC2004 RelWithDebInfo LVI`, and will always be ran.
* Removed EEID test `ACC2004 clang-11 RelWithDebInfo ControlFlow EEID` from Azure-Linux

# Changed Windows Tests
* Packaging `Windows 2019 Debug LVI` -> Azure-Windows `XC Win2019 v2 clang-11 Debug ControlFlow`. This is a pre-existing test, and will run only in the full test suite.
* Packaging `Windows 2019 RelWithDebInfo` -> Azure-Windows `XC Win2019 v2 RelWithDebInfo`. This is a new test, and will run only in the full test suite.
* Packaging `Windows 2019 RelWithDebInfo LVI` -> Azure-Windows `XC Win2019 v2 clang-11 RelWithDebInfo ControlFlow-Clang`. This is a pre-existing test, and will run only in the full test suite.
* Packaging `Windows 2019 Sim Debug LVI` -> Azure-Windows `XC Win2019 v2 clang-11 Debug ControlFlow Sim`. This is a pre-existing test, and will run only in the full test suite.
* Packaging `Windows 2019 Sim RelWithDebInfo` -> Azure-Windows `XC Win2019 v2 RelWithDebInfo Sim`. This is a new test, and will run only in the full test suite.
* Packaging `Windows 2019 Sim RelWithDebInfo LVI` -> Azure-Windows `XC Win2019 v2 RelWithDebInfo ControlFlow Sim`. This is a new test, and will run only in the full test suite.

# Jenkins Library changes
* Deprecated `common.ContainerTasks()` in favor of `common.ContainerRun()`.
* Deprecated parameter `compiler_version` in `common.Run()`
* Changed `helpers.CmakeArgs()` to add boolean arguments `use_snmalloc` and `use_eeid` for `-DUSE_SNMALLOC` and `-DWITH_EEID` flags respectively and removed parameter `cmake_args`. Added logic to validate arguments.
* Changed `tests.ACCTest()` to add arguments `lvi_mitigation`, `lvi_mitigation_skip_tests`, `use_snmalloc`, and `use_eeid` to replace known uses of `extra_cmake_args`.
* Changed `tests.ACCUpgradeTest()` and `tests.ACCContainerTest()` to add arguments `lvi_mitigation`, and `lvi_mitigation_skip_tests` to replace known uses of `extra_cmake_args`.
* Changed `tests.ACCPackageTest()` to add arguments `build_type`. Added arguments `lvi_mitigation`, and `lvi_mitigation_skip_tests` and `use_snmalloc` to replace known uses of `extra_cmake_args`.
* Note: `extra_cmake_args` will be deprecated in other functions soon.
* Added helpers for building with Make. Removed `helpers.InstallBuildCommand()` and added `helpers.makeBuildCommand()`, `helpers.createOpenEnclavePackageCommand()`, `helpers.createHostVerifyPackageCommand()`, `helpers.makeInstallCommand()` and `helpers.ninjaInstallCommand()`
* Removed parameter `hasQuoteProvider` from `tests.WinCompilePackageTest()` and usage thereof in `tests.WinCrossCompile()`
* Removed tests `tests.LinuxPackaging()` and `tests.WindowsPackaging()` in favor of `tests.ACCPackageTest()` and `tests.windowsCrossCompile`

# Text changes
* Added comments and notes to many functions
* Removed Packaging badge from main README

### CODE REVIEW

File: .jenkins/library/vars/tests.groovy [hunk 1]
```diff
@@ -47,11 +55,14 @@ def ACCCodeCoverageTest(String version, String compiler, String build_type) {
...
- *  @param extra_cmake_args: Extra cmake args to pass to the build
+ *  @param lvi_mitigation: Build enclave libraries with LVI mitigation. Choice of: None, ControlFlow-GNU, ControlFlow-Clang, or ControlFlow
+ *  @param lvi_mitigation_skip_tests: Whether to skip LVI mitigation tests. Choice of ON or OFF
+ *  @param use_snmalloc: Whether to use snmalloc allocator. Choice of ON or OFF
+ *  @param use_eeid: Whether to use EEID. Choice of ON or OFF
  *  @param fresh_install: whether to start with a plain VM and install OE dependencies
  */
-def ACCTest(String label, String compiler, String build_type, List extra_cmake_args = [], boolean fresh_install = false) {
-    stage("${label} ${compiler} ${build_type} ${extra_cmake_args} ${fresh_install ? ", e2e" : ""}") {
+def ACCTest(String label, String compiler, String build_type, String lvi_mitigation, String lvi_mitigation_skip_tests = 'OFF', String use_snmalloc = 'OFF', String use_eeid = 'OFF', boolean fresh_install = false) {
```
Comment (CyanDevs) [inline .jenkins/library/vars/tests.groovy:64]: boolean

**Comment by CyanDevs** [comment]
bors try

**Review by mingweishih — APPROVED** [review]

**Review by radhikaj — APPROVED** [review]

**Comment by CyanDevs** [comment]
Override Bors due to repeated errors that are unrelated.

**Pull request merged**